config/7.1.0/securebanking/ig/scripts/groovy/CheckIdempotencyKey.groovy

```diff
@@ -0,0 +1,95 @@
+/*
+ * Script to check the idempotency
+ */
+
+import groovy.json.JsonOutput
+import java.time.Instant
+
+def fapiInteractionId = request.getHeaders().getFirst("x-fapi-interaction-id");
+if (fapiInteractionId == null) fapiInteractionId = "No x-fapi-interaction-id";
+SCRIPT_NAME = "[CheckIdempotencyKey] (" + fapiInteractionId + ") - ";
+
+logger.debug(SCRIPT_NAME + "Running...")
+
+// response object
+responseCheck = new Response(Status.OK)
+responseCheck.headers['Content-Type'] = "application/json"
+
+String idempotencyKeyHeaderValue = request.getHeaders().getFirst("x-idempotency-key")
+logger.debug(SCRIPT_NAME + " idempotency key: " + idempotencyKeyHeaderValue)
+
+def method = request.method
+
+// Only check when post
+switch (method.toUpperCase()) {
+    case "POST":
+        if (idempotencyKeyHeaderValue == null || idempotencyKeyHeaderValue == "") {
+            message = "Failed to get create the resource, 'x-idempotency-key' header / value expected"
+            logger.error(SCRIPT_NAME + message)
+            responseCheck.status = Status.BAD_REQUEST
+            responseCheck.entity = "{ \"error\":\"" + message + "\"}"
+            return responseCheck
+            // the idempotencyKey size must be at most 40 characters
+        } else if(idempotencyKeyHeaderValue.length() > 39){
```

Comment:
`> 40`

Reply:
Fixed condition and all returns with newResultPromise